Sources/PointFree/PageLayout.swift

```diff
@@ -9,49 +9,88 @@ import Prelude
 import Styleguide
 import Tuple
 
-func respond<A>(_ view: View<A>, layout: @escaping (Flash?, View<A>) -> View<A>)
+struct SimplePageLayoutData<A> {
+  private(set) var currentRoute: Route?
+  private(set) var currentUser: Database.User?
+  private(set) var extraStyles: Stylesheet
+  private(set) var data: A
+  private(set) var flash: Flash?
+  private(set) var showTopNav: Bool
+  private(set) var title: String
+  private(set) var useHighlightJs: Bool
+
+  init(
+    currentRoute: Route? = nil,
+    currentUser: Database.User?,
+    data: A,
+    extraStyles: Stylesheet = .empty,
+    showTopNav: Bool = true,
+    title: String,
+    useHighlightJs: Bool = false
+    ) {
+
+    self.currentRoute = currentRoute
+    self.currentUser = currentUser
+    self.data = data
+    self.extraStyles = extraStyles
+    self.flash = nil
+    self.showTopNav = showTopNav
+    self.title = title
+    self.useHighlightJs = useHighlightJs
+  }
+}
+
+func respond<A, B>(
+  view: View<B>,
+  layoutData: @escaping (A) -> SimplePageLayoutData<B>
+  )
   -> Middleware<HeadersOpen, ResponseEnded, A, Data> {
```

Comment:
this is the layout helper. the two generic parameters is what allows this middleware to take all the data needed for the layout to do its thing, while still allowing the view to only take the data it needs to do its job.